app/replica.go

```diff
@@ -137,14 +137,71 @@ func CloneReplica(s *replica.Server, address string, cloneIP string, snapName st
 }
 
 func startReplica(c *cli.Context) error {
+	dir, replicaType, s, address, frontendIP, cloneIP, snapName, size, err := fetchValues(c)
+	if err != nil {
+		return err
+	}
+	err = handleSize(s, size)
+	if err != nil {
+		return err
+	}
+
+	controlAddress, dataAddress, syncAddress, err := handleAddress(address)
+	if err != nil {
+		return err
+	}
+	logrus.Infof("Starting replica having replicaType: %v, frontendIP: %v, size: %v, dir: %v", replicaType, frontendIP, size, dir)
+	logrus.Infof("Setting replicaAddr: %v, controlAddr: %v, dataAddr: %v, syncAddr: %v", address, controlAddress, dataAddress, syncAddress)
 
+	var resp error
+	controlResp, syncResp, rpcResp := handleResp(s, controlAddress, dataAddress)
+	err = checkBool(c, syncAddress, dir, syncResp)
+	if err != nil {
+		return err
+	}
+	if frontendIP != "" {
+		if address == ":9502" {
+			address = "localhost:9502"
+		}
+		go AutoConfigureReplica(s, frontendIP, "tcp://"+address, replicaType)
+	}
+
+	for s.Replica() == nil {
+		logrus.Infof("Waiting for s.Replica() to be non nil")
+		time.Sleep(2 * time.Second)
+	}
+	if replicaType == "clone" && snapName != "" {
+		err := startClone(s, snapName, address, cloneIP)
+		if err != nil {
+			return err
+		}
+	} else {
+		logrus.Infof("Set clone status as NA")
+		if err := s.Replica().SetCloneStatus("NA"); err != nil {
+			logrus.Error("Error in setting the clone status as 'NA'")
+			return err
+		}
+	}
+	select {
+	case resp = <-controlResp:
+		logrus.Fatalf("Rest API exited: %v", resp)
+	case resp = <-rpcResp:
+		logrus.Fatalf("RPC listen exited: %v", resp)
+	case resp = <-syncResp:
+		logrus.Fatalf("Sync process exited: %v", resp)
+	}
+	return resp
+}
+
+func fetchValues(c *cli.Context) (string, string, *replica.Server, string, string, string, string, string, error) {
```

Comment:
returning too many values is not a good approach IMO, so create a structure and initialize the values and then return it to the callee.

Reply:
rename it to getStartConfig()

Reply:
Should the structure contain all the return values 
OR 
 only the string and *replica.Server values, and error is returned separately?

Reply:
error should be returned separately, all those string fields and *replica.Server will be part of a new structure

Reply:
Got it. Thanks for the clarification.